### DISCUSSION TITLE
중급 프로젝트 3주차 회고

### DISCUSSION THREAD

**Discussion by Crack-Egg** [Party-Time-Job/Party-Time-Job#116] [회고록]
## Keep
현재 우리팀이 잘 하고 있어서 계속 유지하면 좋을 것 같은 부분을 작성해주세요.
이시간을 빌려 한주간 프로젝트에서 잘 된 점들을 서로 마구 칭찬해주는 시간을 가져도 좋겠네요.

## Problem
현재 우리팀이 가지고 있는 문제가 있다면 작성해주세요.
문제의 발견은 곧 우리 팀이 더 성장할 수 있다는 의미이니, 문제 발견을 두려워하지말아요:)

## Try
Problem에 적은 것을 해결하기 위해 우리 팀이 다음에 시도해볼만한 것 혹은 Keep 중에서도 더 개선하거나 시도해보고 싶은 것을 작성해보세요. 다음주에는 Try에 적은 것을 바탕으로 더 좋은 협업을 해요 :)

**Comment by DanteSnow** [comment]
## Keep
팀원 모두가 자신이 원래 맡은 역할이 아닌데도, 솔선수범하여 작업들을 맡아서 일정이 수월하게 진행됐습니다.
적당한 잡담과 농담들 덕분에 프로젝트가 즐거웠습니다. 분위기가 너무 좋았습니다.
이번 프로젝트를 통해서 처음 배워가는 것이 정말 많았습니다. 다른 팀원 분들의 지식을 배워가는 느낌이어서 감사했습니다.
프로젝트 첫 시작부터 끝날 때 까지 서로가 서로를 배려하고 존중하는 분위기가 형성되어 있었습니다.

## Problem
추가기능을 많이 넣지 못했다는 것이 많이 아쉬웠습니다. 제가 잠을 줄였다면 가능했을텐데, 피로가 누적이 돼서 그러지 못했습니다.
conflict 해결을 확실하게 하지 못하고 다른 팀원의 작업물을 지운 상태로 머지를 해버렸습니다.
팀원들이 사는 곳이 멀어서 다같이 얼굴을 대면해보지 못했다는 것이 많이 아쉬웠습니다. 다같이 모여서 프로젝트를 진행했다면, 학습적인 부분의 시너지가 더 많이 났을 것 같습니다.

## Try
피로가 누적되지 않게 적절한 휴식을 취할 것입니다.
conflict는 꼼꼼하게 확인하고, 머지를 하기전에도 다시 한 번 더 꼼꼼하게 확인을 하는 습관을 들일 것입니다.

**Comment by Crack-Egg** [comment]
## Keep
정말 3주도 안되는 기간이었지만 이번 프로젝트를 하면서 너무 많은 점을 배울 수 있었던 것 같습니다. 기초 프로젝트 떄에 비해 난이도도 올라가며 전에는 사용해 보지 않았던 폴더 구조와 여러 라이브러리를 사용해 어려움이 있었습니다. 하지만 팀원분들께서 먼저 도와주시려고 하시는 모습과 서로의 문제점을 같이 해결해주시려는 모습으로 제가 중급 프로젝트를 완성할 수 있었다고 생각하고 있습니다. 치현님의 경우 정말 팀장의 표본이라고 할 수 있을 정도로, 팀원들을 잘 이끌어 주셨으며 항상 늦은 시간까지 고민하시고 해결하려고 하셨습니다. 치현님이 이번 프로젝트의 높은 완성도에 정말 큰 역할을 해주셨던 것 같습니다. 시열님은 항상 빠른 속도로 작업을 해주셔서 옆에서 정말 자극을 많이 받을 수 있었던 것 같습니다. 그리고 시열님의 코드를 보면서 많이 배웠던 것 같습니다. 윤성님은 항상 새로운 것에 도전하려고 하시며, 문제점이 생기면 끝까지 해결하시려는 모습을 보여주셨습니다. 그리고 팀원들이 놓칠 수 있었던 사소한 부분들까지 확인해 주셔서 프로젝트가 원활하게 진행되었던 것 같습니다. 이번 프로젝트가 개인적으로 어려웠지만 좋은 팀원들을 만나 끝까지 잘 완성할 수 있었던 것 같습니다!

## Problem
너무 제 페이지에 대한 완성을 우선으로 생각하여 공통으로 해결해야 할 문제들을 너무 팀원분들께 부담드렸던 것 같습니다. 어떻게 보면 모두 같은 작업량을 가지고 프로젝트를 진행했지만 저만 크게 도와드린 게 없던 것 같아 죄송한 부분이 있었습니다. 이로 인해 다른 팀원분들이 늦은 시간까지 작업을 할 떄도 있었던 것 같습니다.

## Try
우선 제 페이지가 완성이 되면 빠르게 공통으로 작업할 수 있는 부분을 확인하고 먼저 해결하려고 하겠습니다. 해결하지 못할 수 있어서 도전을 못했지만, 다음부터는 사소한 부분이라도 먼저 나서서 해결하려고 하겠습니다.

**Comment by siyeol97** [comment]
Keep
- 팀원분들께서 맡은 역할을 끝까지 수행하기위해 노력하는 모습이 좋았습니다
- 말하지 않아도 각자 버그나 부족한 부분을 메꾸려고 하는 모습이 좋았습니다.
- 팀원분들께서 항상 늦은 시간까지 작업을 하시고, 어려운 부분도 솔선수범 하시는 모습이 귀감이 된것 같습니다

Problem
- 시간이 일주일만 더 있었으면 더 많은 기능을 구현하고 완벽한 결과물을 만들 수 있었는데 아쉽습니다.
- 제가 맡은 부분을 너무 완벽하게 하느라 작업 시간이 늦어져 다른 팀원 분들을 많이 도와 드리지 못한게 아쉽습니다.
- 범용성 있는 컴포넌트를 만들지 못해 아쉽습니다.

Try
- 컴포넌트를 만들 때 여러가지 상황을 고려해 범용성 있게 만들겠습니다.
- 생각하고 코드를 작성하는 과정을 단축시키기 위해 더 공부를 열심히 하겠습니다.

**Comment by yunsunglee2** [comment]
## Keep
- 맡은 역할 책임감 있게 해내는 모습이 좋았습니다.
- 어려운 부분이 생기면 서로 도와주며 소통하는 시간이 중요하다고 느꼈습니다.
- 이슈 정리를 깃헙으로 하며 서로의 상황을 알수 있어 좋았습니다

## Problem
- 명세서에 맡게 완벽하게 구현을 못한거 같아 아쉽습니다.
- 프로젝트 명세에 맞춰 작업하느라 숙지가 필요한 부분을 제대로 짚고 가지 못하느 느낌이 들어 마음에 여유가 없었던거 같습니다.
- 마감 기한에 쫓겨 컨벤션 적용이 미흡한거 같습니다 
- 전반적인 작업을 마치고 틀을 만들어 놓은 다음에 리팩토링 하며 작업하는 방법이 작업 능률에 있어 좋다고 느꼈습니다

## Try
- 원활한 작업을 위해 어떻게 해야할지 고민해보고 나중에 리팩토링한다고 너무 밀어두지 말고 작업하는 순간 최대한 공부하고 적용해보려고 노력하겠습니다
